src/EVM.hs

```diff
@@ -852,13 +852,18 @@ exec1 = do
             xValue:xOffset:xSize:xs ->
               accessMemoryRange xOffset xSize $ do
                 availableGas <- use (#state % #gas)
-                let
-                  (cost, gas') = costOfCreate fees availableGas xSize False
-                newAddr <- createAddress self this.nonce
+                let (cost, gas') = costOfCreate fees availableGas xSize False
+
+                -- handle `prank`
+                let from' = fromMaybe self vm.state.overrideCaller
+                resetCaller <- use (#state % #resetCaller)
+                when resetCaller $ assign (#state % #overrideCaller) Nothing
+
+                newAddr <- createAddress from' this.nonce
```

Comment:
should the nonce be from's ? it might not be that important though if it's only off during pranking.

Reply:
I am actually not sure. This: https://book.getfoundry.sh/cheatcodes/prank says it: `Sets msg.sender to the specified address for the next call.` does not talk about nonce. No idea what that means.

Reply:
OK, we decided to leave this be for the moment.